shared/src/commonMain/kotlin/com/github/jetbrains/rssreader/app/FeedStore.kt

```diff
@@ -10,7 +10,10 @@ import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.launch
+import kotlin.js.ExperimentalJsExport
+import kotlin.js.JsExport
 
+@JsExport
 data class FeedState(
     val progress: Boolean,
     val feeds: List<Feed>,
```

Comment:
`List` is not exportable interface, it can create some issues with using it from JavaScript.
I see that in fact you use only `RssReaderJsViewModel` from JavaScript
What about make `observeStore` parameter with type `(state: FeedStateWrapper) -> Unit`
And `FeedStateWrapper` will be something like

```
class FeedStateWrapper(val progress: Boolean, val feeds: List<Feed>)

fun FeedStateWrapper.toFeedState() = FeedState(
    progress,
    feeds.toList()
)
```

It makes all methods of `RssReaderJsViewModel` to be good exported to JS and you can consume it from javascript side with just JS arrays?

Reply:
.. I have tried something .. not really sure tho
also because we need to consume a list of `Post` which is in a `Feed` I came out with that

```kotlin
@ExperimentalJsExport
@JsExport
class FeedStateJsWrapper(
    val progress: Boolean,
    val feeds: Array<FeedJsWrapper>,
    val selectedFeed: Feed? = null
)

@ExperimentalJsExport
@JsExport
class FeedJsWrapper(
    val title: String,
    val link: String,
    val desc: String,
    val imageUrl: String?,
    val posts: Array<Post>,
    val sourceUrl: String,
    val isDefault: Boolean
)

fun FeedState.toJsObject() = FeedStateJsWrapper(
    progress,
    feeds.map {
        Feed(
            it.title,
            it.link,
            it.desc,
            it.imageUrl,
            it.posts,
            it.sourceUrl,
            it.isDefault
        ).toJsObject()
    }.toTypedArray(),
    selectedFeed
)

fun Feed.toJsObject() = FeedJsWrapper(
    title,
    link,
    desc,
    imageUrl,
    posts.toTypedArray(),
    sourceUrl,
    isDefault
)
```

Then we would have that `observeState` method

```kotlin
fun observeState(callback: (state: FeedStateJsWrapper) -> Unit) {
        mainScope.launch {
          store.observeState().collect {
              callback(it.toJsObject())
          }
     }
 }
```

would this be correct ?